pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -3740,13 +3750,20 @@ public void updateArray(String columnName, @Nullable Array x) throws SQLExceptio
                 PSQLState.INVALID_PARAMETER_VALUE);
       }
     } else if (type == OffsetDateTime.class) {
-      if (sqlType == Types.TIMESTAMP_WITH_TIMEZONE || sqlType == Types.TIMESTAMP) {
+      if (sqlType == Types.TIMESTAMP_WITH_TIMEZONE || sqlType == Types.TIMESTAMP || sqlType == Types.TIME) {
```

Comment:
This was actually a bug (new test was added): https://github.com/pgjdbc/pgjdbc/pull/2467/files#diff-7e93771092eab9084402e3c7c81319a1f037febdc7614264329bd29f11d39ef2R690 says it must be supported